### PR TITLE
Improve release monitor

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | See below
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | Download repo, composer install on Linux/WSL, run 
| Possible impacts? | 

### New features
- Live demo - https://danielhlavacek.cz/_files/monitor.html
- Fix CLI output.
- Fix parsing milestone name - used regex for it.
- Report multiple milestones open.
- Check versions in module file and config.xml, report mismatch.
- Remove Corp modules that should not be in that list, it will be solved later.

### How to test
- Download repo, `composer install` on Linux/WSL
- Get token here - https://github.com/settings/personal-access-tokens/new
- Run in web browser using `generate.php?token=yourtoken` or in CLI by `php generate yourtoken`

### How does it look
versions

### CODE REVIEW

**Comment by SharakPL** [comment]
obraz

Maybe show this warning only if there are open PRs available, because you can't know if next version should be a major one or just a patch. Right?

It's similar with multiple milestones. Some PRs may require a major version, others must patch or improve current major version as well. IMO if milestones have different major versions then alert shouldn't appear. Looks like both mentioned repos have multiple milestones for the same major version so in this case alert is correct.

**Comment by Hlavtox** [comment]
Well, I think that we should always have a milestone for a patch version, and rename it if needed, no? We have only one release channel and there will never be PRs merged to multiple milestones.

**Comment by SharakPL** [comment]
You're right 👍

File: generate.php [hunk 1]
```diff
@@ -62,18 +67,21 @@ function getClassByNbCommitsAhead(int $nbCommitsAhead): string
     return 'table-' . $trClass;
 }
 
-echo 'Fetching list of modules</br>';
+echo 'Fetching list of modules' . $separator;
 $modulesToProcess = getModules($client);
-// $modulesToProcess = ['blockreassurance', 'statsstock'];
-echo 'Module list initialized, found ' . count($modulesToProcess) . ' module</br>';
-echo 'Rendering table</br>';
+// $modulesToProcess = ['ps_categoryproducts'];
```
Comment (kpodemski) [inline generate.php]:
```suggestion
```

File: generate.php [hunk 2]
```diff
@@ -184,28 +237,43 @@ function getClassByNbCommitsAhead(int $nbCommitsAhead): string
 
 // Process notifications
 $notifications_html = '';
+if (!empty($notifications['version_mismatch'])) {
+    $notifications_html .= '<div class="alert alert-warning" role="alert">
+        Modules <strong>' . implode(', ', $notifications['version_mismatch']) . '</strong> have different module versions in module file and config.xml, unify it!
```
Comment (kpodemski) [inline generate.php]:
```suggestion
        Modules <strong>' . implode(', ', $notifications['version_mismatch']) . '</strong> have different module versions in the module file and config.xml, it's required to unify it.
```

File: generate.php [hunk 3]
```diff
@@ -184,28 +237,43 @@ function getClassByNbCommitsAhead(int $nbCommitsAhead): string
...
 if (!empty($notifications['unclosed_milestone'])) {
     $notifications_html .= '<div class="alert alert-warning" role="alert">
-        Modules <strong>' . implode(', ', $notifications['unclosed_milestone']) . '</strong> have unclosed milestones, close them.
+        Modules <strong>' . implode(', ', $notifications['unclosed_milestone']) . '</strong> have unclosed milestones, close them. Check that there were no PRs added to these milestones after release.
     </div>';
 }
 if (!empty($notifications['no_next_milestone'])) {
     $notifications_html .= '<div class="alert alert-warning" role="alert">
-        Modules <strong>' . implode(', ', $notifications['no_next_milestone']) . '</strong> don\'t have next milestones.
+        Modules <strong>' . implode(', ', $notifications['no_next_milestone']) . '</strong> don\'t have next milestones. Create them.
```
Comment (kpodemski) [inline generate.php]:
```suggestion
        Modules <strong>' . implode(', ', $notifications['no_next_milestone']) . '</strong> don\'t have next milestones. You should create them.
```

**Comment by Hlavtox** [comment]
Ready to review 🙏

**Review by SharakPL — APPROVED** [review]

**Review by zuk3975 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by Hlavtox** [comment]
yaaaaaaaaaay

**Pull request merged**